src/Analyzer/Passes/FuseFunctionsPass.cpp

```diff
@@ -48,7 +48,7 @@ class FuseFunctionsVisitor : public InDepthQueryTreeVisitor<FuseFunctionsVisitor
             /// Do not apply for `count()` with without arguments or `count(*)`, only `count(x)` is supported.
             return;
 
-        mapping[QueryTreeNodeWithHash(argument_nodes[0])].push_back(&node);
+        mapping[QueryTreeNodeWithHash(argument_nodes[0])].insert(&node);
```

Comment:
Why do we even get any duplicates here? I can't understand

Reply:
Some subtrees can refer to the same node (at least after some optimizations, `AggregateFunctionsArithmericOperations` in the example below replaces both column with the same `avg` function).

```sql
EXPLAIN QUERY TREE run_passes = 1, dump_passes = 1, passes = 4
SELECT avg(b), x - 2 AS b FROM (SELECT number as x FROM numbers(1)) GROUP BY x;


Pass 1 QueryAnalysis - Resolve type for each query expression. Replace identifiers, matchers with query expressions. Perform constant folding. Evaluate scalar subqueries.
Pass 2 NormalizeCountVariants - Optimize count(literal), sum(1) into count().
Pass 3 CustomizeFunctions - Customize implementation of aggregate functions, and in functions.
Pass 4 AggregateFunctionsArithmericOperations - Extract arithmeric operations from aggregate functions.
QUERY id: 0
  PROJECTION COLUMNS
    avg(b) Float64
    b Int64
  PROJECTION
    LIST id: 1, nodes: 2
      FUNCTION id: 2, function_name: minus, function_type: ordinary, result_type: Int64
        ARGUMENTS
          LIST id: 3, nodes: 2
            FUNCTION id: 4, function_name: avg, function_type: aggregate, result_type: Float64
              ARGUMENTS
                LIST id: 5, nodes: 1
                  COLUMN id: 6, column_name: x, result_type: UInt64, source_id: 7
            CONSTANT id: 8, constant_value: UInt64_2, constant_value_type: UInt8
      FUNCTION id: 2, function_name: minus, function_type: ordinary, result_type: Int64
        ARGUMENTS
          LIST id: 3, nodes: 2
            FUNCTION id: 4, function_name: avg, function_type: aggregate, result_type: Float64
              ARGUMENTS
                LIST id: 5, nodes: 1
                  COLUMN id: 6, column_name: x, result_type: UInt64, source_id: 7
            CONSTANT id: 8, constant_value: UInt64_2, constant_value_type: UInt8
  JOIN TREE
    QUERY id: 7, is_subquery: 1
      PROJECTION COLUMNS
        x UInt64
      PROJECTION
        LIST id: 9, nodes: 1
          COLUMN id: 10, column_name: number, result_type: UInt64, source_id: 11
      JOIN TREE
        TABLE_FUNCTION id: 11, table_function_name: numbers
          ARGUMENTS
            LIST id: 12, nodes: 1
              CONSTANT id: 13, constant_value: UInt64_1, constant_value_type: UInt8
  GROUP BY
    LIST id: 14, nodes: 1
      COLUMN id: 6, column_name: x, result_type: UInt64, source_id: 7
```